src/Data/Options.purs

```diff
@@ -14,13 +16,27 @@
 -- |
 -- | and so on.
 -- |
+-- | ## Why PureScript Records Don't Work
+-- |
 -- | PureScript's record types can be a little awkward for this, since it is
 -- | usually the case that any subset of these properties can be specified;
 -- | however, a value of  type `{ flags :: String, defaultEncoding :: String, [...] }`
 -- | must include every property listed, even if you only want to specify
--- | one or two properties.
+-- | one or two properties. If we were to provide a strongly-typed solution,
+-- | we would need a sum type for every possibile combination. For example,
+-- | ```
+-- | data WritStreamOptions
```

Comment:
I think this is a bit of a straw-man argument; there are a number of different approaches of varying effectiveness, but this is probably one of the few which almost nobody would ever seriously consider. For example, a simpler way of handling this using records would be to use Nullable or Undefinable:

```
type Options = { flags :: Nullable String, defaultEncoding :: Nullable String }
```

Admittedly, one issue with that is that including a field with a `null` or `undefined` value is different from not including it at all, and I suspect some JS libraries do distinguish these cases.

Reply:
I think I'll just delete the content I added. A cookbook recipe would probably answer any remaining questions a person had.